addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/DataGenerator.java

```diff
@@ -101,11 +104,11 @@ private String generateParam(String name, String example, Parameter parameter) {
         if (example != null && !example.isEmpty()) {
             return example;
         }
-        if (isArray(parameter.getSchema().getType())) {
+        if (parameter.getSchema() != null && isArray(parameter.getSchema().getType())) {
             return generateArrayValue(name, parameter);
         }
 
-        if (parameter.getSchema() instanceof ArraySchema) {
+        if (parameter.getSchema() != null && parameter.getSchema() instanceof ArraySchema) {
```

Comment:
Seems this is just delaying the inevitable, better add the check before (and return the empty string), instead of letting go through until reaching `getExampleValue`.